modules/custom/inope_policy_review/inope_policy_review.module

```diff
@@ -144,36 +234,284 @@ function inope_policy_review_form($form, &$form_state, $node) {
+    ':nid' => $node->nid,
+    ':uid' => $account->uid,
+  ))->fetchAll();
 
-    drupal_set_message(t('Node ID : @nodeid', array('@nodeid' => $node->nid)));
-    drupal_set_message(t('Node VID : @nodevid', array('@nodevid' => $node->vid)));
-    // TODO: put this into the database.
+  return $policy_review_notes;
+}
+
+/**
+ * Select query for the a single policy review note.
+ */
+function inope_policy_review_select_query_one($policy_review_id) {
+
+  // Select query to get a single record based on the primary key.
+  $policy_review_note = db_query("
+  SELECT * FROM {policy_review} WHERE policy_review_id = :policy_review_id", array(
+    ':policy_review_id' => $policy_review_id,
+  ))->fetchObject();
+
+  return $policy_review_note;
+}
+
+/**
+ * Insert query to add a new policy review note or approve policy.
+ *
+ * Only insert the new row if it is substantially a new request meaning that
+ * some value other than the timestamp is different.
+ */
+function inope_policy_review_insert_query($node, $account, $text, $status) {
+
+  $results = inope_policy_review_select_query($node, $account);
+
+  // If the entry already exists don't duplicate it.
+  if (!empty($results)) {
+    drupal_set_message(t('You have previously submitted a response.'));
+    $policy_review_id = 0;
   }
-  // If a bug is encountered.
   else {
-    drupal_set_message(t('Form state not recognized'));
+    // Run an insert query on the policy review table for the new note.
+    $policy_review_id = db_insert('policy_review')
+      ->fields(array(
+        'nid' => $node->nid,
+        'vid' => $node->vid,
+        'uid' => $account->uid,
+        'note_status' => $status,
+        'note_submitted' => time(),
+        'note' => $text,
+      ))
+      ->execute();
+  }
+
+  return $policy_review_id;
+}
+
+/**
+ * Update query if the user wants to update a record.
+ *
+ * Currently this funtion is not called anywhere but it is avaliable for future
+ * use.
+ */
+function inope_policy_review_update_query(&$form_state) {
+  global $user;
+  $account = $user;
+
+  $note_status = $form_state['values']['inope_policy_review_status_radio'];
+
+  // If the user changed note status to approved, remove the old change request.
+  if ($form_state['values']['inope_policy_review_status_radio'] == 0) {
+    $form_state['values']['inope_policy_review_changes_requested_textarea'] = '';
+  }
+
+  // Run an update query on the policy review table to change the note.
+  $updated_records = db_update('policy_review')
+    ->fields(array(
+      'note_status' => $note_status,
+      'note' => $form_state['values']['inope_policy_review_changes_requested_textarea'],
+    ))
+    ->condition('policy_review_id', $form_state['values']['inope_policy_review_hidden_policy_review_id'], '=')
+    ->execute();
+
+  return $updated_records;
+}
+
+/**
+ * Build a table to display the data for the current node.
+ */
+function inope_policy_review_table($node, $account) {
+  // Set table headers.
```

Comment:
On this table we have an extra column with an edit, we should probably not orphan that column -- lets add a header called "Action" for that column.